Meter.c

```diff
@@ -286,14 +292,38 @@ static const char* const GraphMeterMode_dotsAscii[] = {
    /*20*/":", /*21*/":", /*22*/":"
 };
 
+static void GraphMeterMode_expandGraphData(GraphData* data, unsigned int numRecords) {
+   if (data->numRecords >= numRecords) {
+      assert(numRecords == 0 || data->values != NULL);
+      return;
+   }
+
+   double* buf = malloc(numRecords * sizeof(*data->values));
+   if (!buf)
+      return;
```

Comment:
In the case of an allocation failure the old buffer is still present and nValues is still at its old value. The only case where this might be problematic is if previously there was no data allocated, in which case we need to skip any further processing in the caller (to be checked, but IIRC there was already a check for this).

Reply:
Okay, it looks like that should work as written, but it seems overly subtle and fragile IMO -- I'd still be in favor of switching it to something more robust and harder to misunderstand.